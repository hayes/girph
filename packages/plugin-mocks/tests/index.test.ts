import './examples/starwars/schema';
import { execute } from 'graphql';
import { gql } from 'graphql-tag';
import builder from './examples/starwars/builder';

describe('mocked', () => {
  it('query some stuff', async () => {
    const mockedSchema = builder.toSchema({
      mocks: {
        Character: {
          name: () => 'C-3PO',
        },
      },
    });

    const query = gql`
      query {
        r2d2 {
          name
        }
      }
    `;

    const result = await execute({
      schema: mockedSchema,
      document: query,
      contextValue: {},
    });

    expect(result).toMatchSnapshot();
  });

  describe('type mocks', () => {
    it('queries stuff', async () => {
      const DroidMock = builder.createObjectMock('Droid', () => ({
        name: 'C-3PO',
      }));
      const mockedSchema = builder.toSchema({
        mocks: {},
        typeMocks: [DroidMock],
      });

      const query = gql`
        query {
          r2d2 {
            name
          }
        }
      `;

      const result = await execute({
        schema: mockedSchema,
        document: query,
        contextValue: {},
      });

      expect(result).toMatchSnapshot();
    });

    it('queries lists', async () => {
      const DroidMock = builder.createObjectMock('Droid', () => ({
        type: 'Droid' as const,
        name: 'C-3PO',
        friends: ['1002', '1003'],
      }));
      const mockedSchema = builder.toSchema({
        mocks: {},
        typeMocks: [DroidMock],
      });

      const query = gql`
        query {
          r2d2 {
            name
            friends {
              name
            }
          }
        }
      `;

      const result = await execute({
        schema: mockedSchema,
        document: query,
        contextValue: {},
      });

      expect(result).toMatchSnapshot();
    });
  });
});
