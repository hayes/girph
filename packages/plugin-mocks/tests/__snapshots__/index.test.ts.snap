// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mocked query some stuff 1`] = `
Object {
  "data": Object {
    "r2d2": Object {
      "name": "C-3PO",
    },
  },
}
`;

exports[`mocked type mocks queries lists 1`] = `
Object {
  "data": Object {
    "r2d2": Object {
      "friends": Array [
        Object {
          "name": "C-3PO",
        },
      ],
      "name": "C-3PO",
    },
  },
}
`;

exports[`mocked type mocks queries stuff 1`] = `
Object {
  "data": Object {
    "r2d2": Object {
      "name": "C-3PO",
    },
  },
}
`;
